neurom/features/neurite.py

```diff
@@ -74,8 +74,13 @@ def homogeneous_filter(section):
 
     # forking sections cannot be heterogeneous
     if (
-        iterator_type in {Section.ibifurcation_point, Section.iforking_point}
-        and section_type != NeuriteType.all
+        section_type != NeuriteType.all
+        and not SubtypeCollection(section_type).is_composite()
```

Comment:
Is this expensive to compute?

Reply:
Yeah, I was referring to the ctor. I am also wondering why do we need to check the section_type for being a composite type. It is counter-intuitive for a section type to be a composite imho.

Reply:
In some cases we pass the `neurite_type` as `section_type` in the heterogeneous case of `iter_neurites`, so a composite type can arrive here. It just mean we want to filter the sections according to the subtypes of the composite type, not that the filtered sections must have a composite type. But maybe we should make it more clear in the docs?

Reply:
In the case of composites, why not pass NeuriteType.all to the section_type if the neurite_type is composite and process_subtrees=True? In the end, as far as section filtering is concerned, when a composite type matches a neurite all the sections are traversed, isn't that the case?

Reply:
The only thing we want here is to keep bifurcation points when a composite type is given with `ibifurcation_point` or
`iforking_point` iterator, so I think it's better to keep it here.